app/celery/reporting_tasks.py

```diff
@@ -67,21 +67,23 @@ def create_nightly_billing_for_day(process_day):
 @notify_celery.task(name="create-nightly-notification-status")
 @cronitor("create-nightly-notification-status")
 @statsd(namespace="tasks")
-def create_nightly_notification_status(day_start=None):
-    # day_start is a datetime.date() object. e.g.
-    # 4 days of data counting back from day_start is consolidated
-    if day_start is None:
-        day_start = convert_utc_to_bst(datetime.utcnow()).date() - timedelta(days=1)
-    else:
-        # When calling the task its a string in the format of "YYYY-MM-DD"
-        day_start = datetime.strptime(day_start, "%Y-%m-%d").date()
+def create_nightly_notification_status():
+    most_recent_day_to_process = convert_utc_to_bst(datetime.utcnow()).date() - timedelta(days=1)
+
     for i in range(0, 4):
-        process_day = day_start - timedelta(days=i)
+        process_day = most_recent_day_to_process - timedelta(days=i)
         for notification_type in [SMS_TYPE, EMAIL_TYPE, LETTER_TYPE]:
```

Comment:
Very optional: You could exclude letters from this loop and do `range(10)` on the second loop instead. Means behaviour for letters isn't spread across two loops

Reply:
yeah i think that's a marginally nicer way of laying out the function 👍. squashed in